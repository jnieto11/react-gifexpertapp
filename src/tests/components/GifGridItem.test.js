import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
configure({ adapter: new Adapter() });


describe('Pruebas en <GifGridItem></GifGridItem>', () => {
    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    test('Debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe tener un parrafo con el title', () => {
        
        const p  = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    })
    
    test('Debe tener la imagen  igual al url y alt de los props con el title', () => {
        
        const img  = wrapper.find('img');     
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    })
    
    test('Debe tener animate__fadeIn', () => {
        
        const div  = wrapper.find('div');     
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })

})
