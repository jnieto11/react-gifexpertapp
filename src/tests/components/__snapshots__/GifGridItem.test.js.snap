// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem></GifGridItem> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    className="img"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
